libraries/AP_PacketDigitalCAN/AP_PacketDigitalCAN.h

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2020 Kraus Hamdani Aerospace, Inc. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Tom Pittenger
+ *
+ */
+
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_CANManager/AP_CANDriver.h>
+
+#ifndef HAL_PACKETDIGITAL_CAN_ENABLE
+    #define HAL_PACKETDIGITAL_CAN_ENABLE (HAL_NUM_CAN_IFACES && !HAL_MINIMIZE_FEATURES)
+#endif
+
+#if HAL_PACKETDIGITAL_CAN_ENABLE
+#include <AP_HAL/Semaphores.h>
+#include <AP_Param/AP_Param.h>
+
+#define PACKETDIGITAL_TIMEOUT_MS    2000
+
+
+class AP_PacketDigitalCAN : public AP_CANDriver
+{
+public:
+    AP_PacketDigitalCAN();
+    ~AP_PacketDigitalCAN();
+
+    /* Do not allow copies */
+    AP_PacketDigitalCAN(const AP_PacketDigitalCAN &other) = delete;
+    AP_PacketDigitalCAN &operator=(const AP_PacketDigitalCAN&) = delete;
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    // Return AP_PacketDigitalCAN from @driver_index or nullptr if it's not ready or doesn't exist
+    static AP_PacketDigitalCAN *get_pdcan(uint8_t driver_index);
+
+    // initialize AP_PacketDigitalCAN bus
+    void init(uint8_t driver_index, bool enable_filters) override;
+    bool add_interface(AP_HAL::CANIface* can_iface) override;
+
+    // test if the AP_PacketDigital CAN driver is ready to be armed
+    bool pre_arm_check(char* reason, uint8_t reason_len) const;
+
+    bool get_voltage_and_current_and_temp(const uint16_t serialnumber, const bool chkInput, float &voltage, float &current, float &temperature) const;
+
+    bool is_healthy(const uint16_t serialnumber) const {
+        return is_healthy_by_index(get_device_index(serialnumber));
+    }
+
+private:
+
+    // Message groups form part of the CAN ID of each frame
+    enum class PacketType : uint8_t {
+        UNKNOWN         = 0,
+        ACK             = 0x21,
+        NACK            = 0x22,
+        PING            = 0x23,
+        FAULT           = 0x42,
+        ALGORITHM_SET   = 0x44,
+        ALGORITHM_GET   = 0x45,
+        STREAM_FAULT    = 0x47,
+        STREAM_INPUT    = 0x48,
+        STREAM_OUTPUT   = 0x49,
+        VOLTAGE_SET     = 0x4B,
+        VOLTAGE_GET     = 0x4C,
+        CVT_SET         = 0x4D,
+        CVT_GET         = 0x4E,
+    };
+
+    enum class Fault_Flags : uint8_t {
+        NONE                = 0,
```

Comment:
Remove this.  `NONE` is not a flag, and you check a mask against zero, not against a specific flag value from the enumeration.

Reply:
checking against zero does not work. Either way, I've removed it